### PR TITLE
Be able to parse CSV from STDIN

### PR DESCRIPTION
**Pull request by miya0001**
Related:
https://github.com/wp-cli/entity-command/issues/96

### CODE REVIEW

**Comment by miya0001** [comment]
@wp-cli/committers 
Do you know how test STDIN with `phpunit`?

**Comment by gitlost** [comment]
For just unit testing `Iterators\CSV`, exec is your man:
```php
<?php

class IteratorsTest extends PHPUnit_Framework_TestCase {
	
	function testIteratorsCSVStdin() {

		$php = "require '" . dirname( __DIR__ ) . "/php/WP_CLI/Iterators/CSV.php';\n";
		$php .= <<<'EOD'
$iterator = new WP_CLI\Iterators\CSV( '-' );
foreach( $iterator as $i => $line ) {
	echo "$i";
	foreach ( $line as $j => $field ) echo ":$field";
	echo "\n";
}
EOD;

		$stdin = <<<'EOD'
user_login,user_email,display_name,role
stdinuser1,bobjones@example.com,Bob Jones,contributor
stdinuser2,newuser1@example.com,New User,author
existinguser,existinguser@example.com,Existing User,administrator
EOD;

		$stdin = escapeshellarg( $stdin );
		$php = escapeshellarg( $php );

		$output = array();
		exec( "echo $stdin | php -r $php", $output );

		$this->assertSame( 3, count( $output ) );
		$this->assertSame( '0:stdinuser1:bobjones@example.com:Bob Jones:contributor', $output[0] );
	}
}
```

File: php/WP_CLI/Iterators/CSV.php [hunk 1]
```diff
@@ -18,7 +18,12 @@ class CSV implements \Iterator {
 	private $currentElement;
 
 	public function __construct( $filename, $delimiter = ',' ) {
-		$this->filePointer = fopen( $filename, 'r' );
+		if ( '-' === $filename && $this->has_stdin() ) {
+			$this->filePointer = fopen( 'php://stdin', 'r' );
+		} else {
+			$this->filePointer = fopen( $filename, 'r' );
+		}
```
Comment (danielbachhuber) [inline php/WP_CLI/Iterators/CSV.php]: I think it's too magical (also known as unexpected) to interpret `-` as STDIN.

For the purposes of https://github.com/wp-cli/entity-command/issues/96, we should simply read STDIN to memory, and then parse as CSV.

**Comment by miya0001** [comment]
@gitlost Thanks, you always help me a lot. 😊 
@danielbachhuber OK, I opened new PR. https://github.com/wp-cli/entity-command/pull/100

Thanks again!

**Pull request closed without merging**